imports/wasi_http/http_test.go

```diff
@@ -16,14 +17,45 @@ import (
 )
 
 type handler struct {
-	urls []string
+	urls   []string
+	bodies []string
+}
+
+func deepEquals(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	} else {
+		for ix := range a {
+			if a[ix] != b[ix] {
+				return false
+			}
+		}
+	}
+	return true
+}
```

Comment:
Any reason not to use `reflect.DeepEqual` instead?

Alternatively, more modern Go is trending towards solutions like `slices.Equal` that could be useful here https://pkg.go.dev/golang.org/x/exp/slices#Equal